### ISSUE TITLE
Keycloak: Change 'code' access token request key, to 'pre-authorized_code'

### ISSUE THREAD

**Issue by hugoib** [adorsys/keycloak-ssi-deployment#42]
For the demo, the following change was made inside the wallet:

![Image](https://github.com/adorsys/keycloak-ssi-deployment/assets/4196617/41e42902-841f-40c6-ae4e-498401944179)

Keycloak receives 'code' instead of 'pre-authorized-code'

![Image](https://github.com/adorsys/keycloak-ssi-deployment/assets/4196617/d998c3ff-09c0-4346-a710-a7d5cd8a75c1)

Review the spec and fix where necessary.

**Comment by Marcjazz** [comment]
Here the section of the specs that addresses this issue 
https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#section-4.1.1-4.2.1

**Comment by Marcjazz** [comment]
@francis-pouatcha I've were the problem comes from. We read the `code` instead of the `pre-authorized_code` for the pre-authorized code grant type  

```java
        String code = formParams.getFirst(OAuth2Constants.CODE);
```
https://github.com/adorsys/keycloak-oid4vc/blob/e3f8e70f61cf7f41d6b9c79bfac8af46e781df1b/services/src/main/java/org/keycloak/protocol/oidc/grants/PreAuthorizedCodeGrantType.java#L53

**Comment by francis-pouatcha** [comment]
fixed with commit #5f5f2d1475639968970fd62f7b53f20a5723d5e7

**Comment by francis-pouatcha** [comment]
Resolved with: 
- https://github.com/keycloak/keycloak/pull/30692
